pkg/types/vsphere/validation/platform.go

```diff
@@ -60,12 +62,24 @@ func ValidateForProvisioning(p *vsphere.Platform, fldPath *field.Path) field.Err
 		allErrs = append(allErrs, field.Required(fldPath.Child("network"), "must specify the network"))
 	}
 
-	allErrs = append(allErrs, validateVIPs(p, fldPath)...)
+	if strings.Join([]string{p.APIVIP, p.IngressVIP}, "") != "" {
+		allErrs = append(allErrs, validateVIPs(p, network, fldPath)...)
+	}
```

Comment:
This check **should not** be done here. In `ValidateForProvisioning`, the two VIPs are required. The suggestion that I was trying to make was to only validate that both VIPs were provided here rather than doing all of the other validation for the VIPs, since the other validation will necessarily have already been done.
```suggestion
	if p.APIVIP == "" {
		// add error
	}
	if p.IngressVIP == "" {
		// add error
	}
```